server/authz/errors.go

```diff
@@ -78,6 +78,11 @@ func (e *CheckMissing) Error() string {
 	return ForbiddenErrorMessage
 }
 
+// StatusCode implements the go-kit http StatusCoder interface.
+func (e *CheckMissing) StatusCode() int {
+	return http.StatusForbidden
+}
```

Comment:
Actually we use this "Missing authorization check" error on purpose to make sure we never forget to add authorization (`svc.authz.Authorize`) checks on all APIs.

Reply:
Yes, makes sense. This specific change simply changes the status code from 500 to 403, so it shouldn't affect the intention of "Missing authorization check".

Reply:
Let's discuss.

The reason we decided to explicitly return 500s was to treat developers forgetting an authorization check as a security/serious bug.

@mna may have more insights.

Reply:
I think leaving it as 500 would be better - it would raise an alert if it happened in prod so we know about it, and it's more likely that we'd catch it in integration tests where we e.g. expect a 4xx and find out that a 500 is returned instead. I think receiving a 403 could make it harder to detect those cases.